### DISCUSSION TITLE
How do I do conditional formatting with the tables bubble

### DISCUSSION THREAD

**Discussion by bashbunni** [charmbracelet/bubbles#305] [Q&A] [answered]
[From Discord]

tomekz
OP
 — 12/15/2022 6:31 AM
Hey, I would like to have the values in my [table](https://github.com/charmbracelet/bubbletea/tree/master/examples/table) coloured differently depending on some condition.
So e.g. if the value is negative - show it in red. 
If it's positive -show it in green

Any ideas/examples how to implement such simple feature? 

---------

Solution: https://gist.github.com/bashbunni/c003657992bdb960da87d009d9e95bd9

**Comment by bashbunni** [comment, marked as answer]
The way to do this would be to create some custom styles for the table, then use the table's `SetStyles` function to change the styling based on some condition. You can see how I implemented this in the gist below. In this case it changes the background colour to red when the name of the city is Cairo. 

Solution: https://gist.github.com/bashbunni/c003657992bdb960da87d009d9e95bd9

**Reply by bashbunni** [reply]
Note: there is a table provided by Lip Gloss now that's great to work with and easier to style than the original solution I provided as the answer to this question. Here's the API and links to the example usage.
https://pkg.go.dev/github.com/charmbracelet/lipgloss/table
https://github.com/charmbracelet/lipgloss/tree/master/examples/table

**Comment by felipeospina21** [comment]
In the example, the whole row style changes based on the condition and only when it is selected. I was wondering if there is a way to change the style for just one cell and without the need for it to be selected. So in the example if any row has a cell with a value "Cairo" the word will have a different color. Is this possible with the Table bubble? or a custom implementation would be necessary?

**Reply by bashbunni** [reply]
Hi! I will take a look at this tomorrow - I've been in Lip Gloss table land for a while now, so need to do a quick refresh on the current bubbles rendering for tables

**Comment by KevM** [comment]
There is [a PR I made](https://github.com/charmbracelet/bubbles/pull/504) to address this. Please take a look to see if it helps with your concern.
